src/ldp/representation/RepresentationMetadata.ts

```diff
@@ -1,47 +1,219 @@
+   * @param object - Value to add.
    */
-  raw: Quad[];
+  public add(predicate: NamedNode | string, object: NamedNode | Literal | string): this {
+    this.store.addQuad(this.id, this.convertPredicate(predicate), this.convertObject(object));
+    return this;
+  }
+
   /**
-   * Optional metadata profiles.
+   * Removes the given value from the metadata. Strings get converted to literals.
+   * @param predicate - Predicate linking identifier to value.
+   * @param object - Value to remove.
    */
-  profiles?: string[];
+  public remove(predicate: NamedNode | string, object: NamedNode | Literal | string): this {
+    this.store.removeQuad(this.id, this.convertPredicate(predicate), this.convertObject(object));
+    return this;
+  }
+
   /**
-   * Optional size of the representation.
+   * Removes all values linked through the given predicate.
+   * @param predicate - Predicate to remove.
    */
-  byteSize?: number;
+  public removeAll(predicate: NamedNode | string): this {
+    this.removeQuads(this.store.getQuads(this.id, this.convertPredicate(predicate), null, null));
+    return this;
+  }
+
   /**
-   * Optional content type of the representation.
+   * Finds all object values matching the given predicate.
+   * @param predicate - Predicate to get the values for.
+   *
+   * @returns An array with all matches.
    */
-  contentType?: string;
+  public getAll(predicate: NamedNode | string): Term[] {
+    return this.store.getQuads(this.id, this.convertPredicate(predicate), null, null)
+      .map((quad): Term => quad.object);
+  }
+
   /**
-   * Optional encoding of the representation.
+   * @param predicate - Predicate to get the value for.
+   *
+   * @throws Error
+   * If there are multiple matching values.
+   *
+   * @returns The corresponding value. Undefined if there is no match
    */
-  encoding?: string;
+  public get(predicate: NamedNode | string): Term | undefined {
+    const terms = this.getAll(predicate);
+    if (terms.length === 0) {
+      return;
+    }
+    if (terms.length > 1) {
+      throw new Error(`Multiple results for ${typeof predicate === 'string' ? predicate : predicate.value}`);
+    }
+    return terms[0];
+  }
+
   /**
-   * Optional language of the representation.
+   * Sets the value for the given predicate, removing all other instances.
+   * @param predicate - Predicate linking to the value.
+   * @param object - Value to set.
    */
-  language?: string;
+  public set(predicate: NamedNode | string, object: NamedNode | Literal | string): this {
+    this.removeAll(predicate);
+    this.add(predicate, object);
+    return this;
+  }
+
   /**
-   * Optional timestamp of the representation.
+   * Converts the incoming predicate to a named node.
+   * In case of string, first checks if it is a shortcut, if not a new named node gets made.
+   * @param predicate - Predicate to potentially transform.
    */
-  dateTime?: Date;
+  private convertPredicate(predicate: NamedNode | string): NamedNode {
```

Comment:
```suggestion
  private getPredicateTerm(predicate: NamedNode | string): NamedNode {
```

("convert to what?")